presto-base-jdbc/src/main/java/com/facebook/presto/plugin/jdbc/JdbcRecordSink.java

```diff
@@ -130,6 +132,11 @@ public void appendLong(long value)
                 long localMillis = ISOChronology.getInstanceUTC().getZone().getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
                 statement.setDate(next(), new Date(localMillis));
             }
+            else if (TIMESTAMP.equals(columnTypes.get(field))) {
+                long utcMillis = TimeUnit.MILLISECONDS.toMillis(value);
+                long localMillis = ISOChronology.getInstanceUTC().getZone().getMillisKeepLocal(DateTimeZone.getDefault(), utcMillis);
+                statement.setTimestamp(next(), new Timestamp(localMillis));
```

Comment:
try to use the following method

`statement.setTimestamp(next(), new Timestamp(utcMillis), Calendar.getInstance(/*force time zone to be UTC*/TimeZone.getTimeZone("UTC"), /*force calendar to be Gregorian*/ ENGLISH);)`